src/embedded-checkout/iframe-event-poster.ts

```diff
@@ -0,0 +1,14 @@
+export default class IframeEventPoster<TEvent> {
+    constructor(
+        private _targetOrigin: string,
+        private _targetWindow: Window
+    ) {}
+
+    post(event: TEvent): void {
+        if (window === this._targetWindow) {
```

Comment:
should this check be performed at construction time?

Reply:
Hrmm, good point. I guess what I really want is a no-op messenger. So it does nothing if it is not used an iframe.

Reply:
yeah, I was mostly thinking if there could be a way that no checking for it could lead to a bug, so better throw an exception when constructing it.